src/app/pages/home/exchange/exchange.page.ts

```diff
@@ -152,11 +169,156 @@ export class ExchangePage implements OnInit {
           console.log(error);
         } else {
           console.log("Card: ", data)
+          this.isCardFound = true;
           this.card = data;
           this.frontImg = this.card.frontImg;
           this.backImg = this.card.backImg;
+          this.ownerId = this.authService.getCurrentUserId();
+          console.log('Current owner ID: ', this.ownerId);
+          this.holderId = this.card.holders;
+          console.log('Current holder ID: ', this.holderId);
         }
       });
     }
   }
+
+  async selectCard() {
+    const modal = await this.modalController.create({
+      component: CardSelectPage
+    });
+
+    modal.onWillDismiss().then(data => {
+       console.log(data);
+    });
+    return await modal.present();
+  }
+
+  cancel() {
+    !this.isCardFound;
+    this.router.navigate(['/home/exchange']);
+  }
+
+  startNFCListener() {
+
+    this.nfc.addNdefListener(() => {
+      console.log('successfully attached ndef listener');
+
+      let toast = this.toastCtrl.create({
+        message: 'Connected',
+        duration: 1000,
+        position: 'top'
+      }).then(toast => toast.present());
+
+    }, (err) => {
+      console.log('error attaching ndef listener', err);
+
+      let toast = this.toastCtrl.create({
+        message: err,
+        duration: 1000,
+        position: 'top'
+      }).then(toast => toast.present());
+
+     }).subscribe(async (event) => {
+
+       let ndefMessage = [
+           this.ndef.textRecord(this.ownerId),
+           this.ndef.textRecord(this.id)
+       ];
+
+       this.tagId = "";
+       this.tagOwnerId = "";
+       this.tagCardId = "";
+
+       let tagIdInfo = await this.nfc.bytesToHexString(event.tag.id);
+       this.tagId = tagIdInfo;
+       console.log('decoded tag id', this.nfc.bytesToHexString(event.tag.id));
+
+
+       if (event.tag.ndefMessage){
+          console.log('received ndefMessage. the tag contains: ', event.tag);
+          let payloadIndexOne = event.tag.ndefMessage[0].payload;
+          let tagContentOwnerId = await this.nfc.bytesToString(payloadIndexOne).substring(3);
+          this.tagOwnerId = tagContentOwnerId;
+
+          let payloadIndexTwo = event.tag.ndefMessage[1].payload;
+          let tagContentCardId = await this.nfc.bytesToString(payloadIndexTwo).substring(3);
```

Comment:
Is this something we could bundle with line 232? 
ex:
```
getTagEventPayload(event) {
   const id = await this.nfc.bytesToHexString(event.tag.id);
   const contentOwerId = await this.nfc.bytesToString(payloadIndexOne).substring(3);
   const contentCardId = await this.nfc.bytesToString(payloadIndexTwo).substring(3);
   return { id, contentOwnerId, contentCardId }
}
``

Reply:
That way you can await on this returned object before continuing. Your data should be loaded by then, I think?
```
const tagEventPayload = await getTagEventPayload(event);
if(tagEventPayload) {
 //Share NFC
//Send Card Approval
}

Reply:
Yeah, I see what you're saying. Good thinking! I'll try that this week. I wonder if the data is even available to `return` anything tangible before connecting with the other device. I'll guess we'll see.